src/main/java/ch/njol/skript/variables/Variables.java

```diff
@@ -474,6 +478,67 @@ public static Object getVariable(String name, @Nullable Event event, boolean loc
 		}
 	}
 
+	/**
+	 * Returns an iterator over the values of this list variable.
+	 *
+	 * @param name the variable's name. This must be the name of a list variable, ie. it must end in *.
+	 * @param event if {@code local} is {@code true}, this is the event
+	 *                 the local variable resides in.
+	 * @param local if this variable is a local or global variable.
+	 * @return an {@link Iterator} of {@link Pair}s, containing the {@link String} index and {@link Object} value of the
+	 * 			elements of the list. An empty iterator is returned if the variable does not exist.
+	 */
+	public static Iterator<Pair<String, Object>> getVariableIterator(String name, boolean local, @Nullable Event event) {
+		assert name.endsWith("*");
+		Object val = getVariable(name, event, local);
+		String subName = StringUtils.substring(name, 0, -1);
+
+		if (val == null)
+			return new EmptyIterator<>();
+		assert val instanceof TreeMap;
+		// temporary list to prevent CMEs
+		@SuppressWarnings("unchecked")
+		Iterator<String> keys = new ArrayList<>(((Map<String, Object>) val).keySet()).iterator();
+		return new Iterator<>() {
+			@Nullable
+			private String key;
+			@Nullable
+			private Object next = null;
+
+			@Override
+			public boolean hasNext() {
+				if (next != null)
+					return true;
+				while (keys.hasNext()) {
+					key = keys.next();
+					if (key != null) {
+						next = Variable.convertIfOldPlayer(subName + key, local, event, Variables.getVariable(subName + key, event, local));
```

Comment:
unfortunate... we should aim to add an api for variable conversion (or find a way to move this) (not in this pr of course)